packages/cli/test/graphql.test.ts

```diff
@@ -0,0 +1,69 @@
+import { execa } from 'execa'
+
+describe('graphql', () => {
+  describe('graphql:schema', () => {
+    test('printing graphql schema succeeds', async () => {
+      const schema = await execa('glaze', [
+        'graphql:schema',
+        'test/mocks/runtime.composite.picture.post.json',
+      ])
+      expect(schema.stdout.toString()).toMatchSnapshot()
+    }, 60000)
+
+    test('printing graphql schema succeeds with --readonly flag', async () => {
+      const schema = await execa('glaze', [
+        'graphql:schema',
+        'test/mocks/runtime.composite.picture.post.json',
+        '--readonly',
+      ])
+      expect(schema.stdout.toString()).toMatchSnapshot()
+    }, 60000)
+  })
+
+  describe('graphql:server', () => {
+    test('graphql server starts', async () => {
+      expect.assertions(2)
+      try {
+        await execa(
+          'glaze',
+          ['graphql:server', 'test/mocks/runtime.composite.picture.post.json', '--port=62433'],
+          {
+            timeout: 15000,
+          }
+        )
+      } catch (e) {
+        expect((e as Error).message.includes('Timed out')).toBe(true)
```

Comment:
Why is this expected please?

Reply:
So what I think I should have done in these tests was something like:

```
cont serverProcess = execa(....)
// 1. wait for a few seconds or figure out how to listen to the serverProcess's stdout
// 2. check serverProcess's stdout to see that the message about server running has been printed
serverProcess.kill()
```
, but I didn't figure out how to do 1. and 2., so instead I passed the `timeout` parameter to `execa(...)` here:
```
await execa(
    'glaze',
     ['graphql:server', 'test/mocks/runtime.composite.picture.post.json', '--port=62433'],
     {
        timeout: 15000,
      }
)
```
and I used `await`. The way it works is that `execa` runs the child process for 15s and then kills it with the time out message. That's why the timeout message is expected.


I thought that these tests are good enough for now, as we're gonna have to come up with something better very soon anyways, but I can also work on improving them, before merging this PR, if you don't feel comfortable with doing this :) .

Reply:
Yes ideally the command would handle clean exits, and the test could cover this behavior?
See https://github.com/ceramicstudio/js-glaze/blob/feat/composites-integration/examples/webpack-notes/scripts/start-server.mjs#L27-L31 for example on how to stop the server.

Reply:
Thanks!